src/storage-types/src/configuration.rs

```diff
@@ -33,18 +37,33 @@ pub struct StorageConfiguration {
     /// TODO(guswynn): `ConnectionContext` also contains some shared global state that should
     /// eventually be moved up to this struct.
     pub connection_context: ConnectionContext,
+
+    /// A clone-able `mz_dyncfg::ConfigSet` used to access dyncfg values.
+    config_set: Arc<ConfigSet>,
 }
 
 impl StorageConfiguration {
     /// Instantiate a new `StorageConfiguration` with default parameters and the given context.
-    pub fn new(connection_context: ConnectionContext) -> StorageConfiguration {
+    pub fn new(
+        connection_context: ConnectionContext,
+        config_set: ConfigSet,
+    ) -> StorageConfiguration {
         StorageConfiguration {
             parameters: Default::default(),
             connection_context,
+            config_set: Arc::new(config_set),
         }
     }
 
-    pub fn update(&mut self, parameters: StorageParameters) {
-        self.parameters.update(parameters)
+    /// Get a reference to the shared `ConfigSet`.
+    pub fn config_set(&self) -> &ConfigSet {
+        &self.config_set
+    }
+
+    pub fn update(&mut self, mut parameters: StorageParameters) {
+        if let Some(updates) = parameters.dyncfg_updates.take() {
+            updates.apply(&self.config_set);
+        }
```

Comment:
can you leave a comment explaining this? It's just a non-obvious API since it would seem that the `self.parameters.update` call should handle applying any nested updates